elastic/kibana
# [FTR] [deployment agnostic tests] move saml roles setup to kbn/es

## Summary

closes #190212

This PR moves realm roles and mapping creation from FTR service to `kbn/es` package in oder to configure ES properly when you start servers with  ` node scripts/functional_tests_server.js --config <path/to/stateful_config_with_SAML_setup>`

It allows to start servers and investigate test scenario manually (e.g. reproduce failure)

Thanks a lot to @azasypkin for ideas how to simplify and minimize the code change.

Instead of 2 API calls to ES in FTR service we copy roles.yml to ES config directory before its start and create role mapping via API (but using template) as a part of security setup

How to test: 
```
node scripts/functional_tests_server.js --config x-pack/test/api_integration/deployment_agnostic/stateful.config.ts
curl -X GET "http://localhost:9220/_security/role_mapping" -H "Content-Type: application/json" -u elastic:changeme
```

you should get the valid mapping back

Note: if you try to load Kibana manually, it won't work due to missing mock-idp-plugin in stateful #190221 But automated tests are functional and can be run via script.
